lib/iris/experimental/ugrid/__init__.py

```diff
@@ -3643,7 +3643,7 @@ def _build_connectivity(connectivity_var, file_path, location_dims):
     indices_data = netcdf._get_cf_var_data(connectivity_var, file_path)
 
     cf_role = connectivity_var.cf_role
-    start_index = connectivity_var.start_index
+    start_index = getattr(connectivity_var, "start_index", None)
```

Comment:
Either:

* The default here should be `0`.
* Creation of a `Connectivity` instance should only pass a `start_index` kwarg if `start_index != 1`.
* The `Connectivity` class should be refactored to make `start_index` an optional parameter, applying the default `0` value **within** `__init__()` rather than making it a parameter default.
  (I'm guessing this is the preferred option?)

Reply:
But, what is wrong with passing `None` to indicate 'use default' ?
The use of 'None' as a default is a public in the signature of the Connectivity constructor -- so it is _equivalent_  to not specifying start_index, and you can trust that.
We then just need to trust that Connectivity implements its default in line with the UGRID rules.  Which seem perfectly reasonable to me.

Reply:
Yes, I thought that's exactly what I said in the third bullet?

Reply:
> Yes, I thought that's exactly what I said in the third bullet?
> I thought that's exactly what I said in the third bullet?

Sorry, you're right.  I had mis-read it somehow.
We already have a 'start_index=0', and can't pass a None.
So I would _instead_ fix as your first bullet "default here should be 0"

Reply:
```suggestion
    start_index = getattr(connectivity_var, "start_index", 0)
```